iroh/src/endpoint.rs

```diff
@@ -634,68 +668,42 @@ impl Endpoint {
                 )
             })?;
 
-        debug!(
-            "connecting to {}: (via {} - {:?})",
-            node_id, addr, direct_addresses
-        );
+        debug!("connecting via {addr} - {direct_addresses:?}",);
+
+        let transport_config = options
+            .transport_config
+            .unwrap_or(self.static_config.transport_config.clone());
 
         // Start connecting via quinn. This will time out after 10 seconds if no reachable
         // address is available.
-        self.connect_quinn(node_id, alpn, addr, transport_config)
-            .await
-    }
 
-    #[instrument(
-        name = "connect",
-        skip_all,
-        fields(
-            me = %self.node_id().fmt_short(),
-            remote_node = node_id.fmt_short(),
-            alpn = %String::from_utf8_lossy(alpn),
-        )
-    )]
-    async fn connect_quinn(
```

Comment:
`connect_quinn` used to only be called by `connect_with`.
Instead of separating out the function, I've moved it in with `connect_with`, which was renamed to `connect_with_opts`.

I've tried to preserve the tracing instrumentation though, thus all of the `span.record` calls.